pep/app/services/current-user.ts

```diff
@@ -263,59 +264,79 @@ export default class CurrentUserService extends Service {
     ): Promise<Result<UserPreferences | undefined, UserPreferenceError>> {
         if (this.user?.userType === UserType.GROUP || !this.session.isAuthenticated) {
             this.informationBar.show('settings-auth');
-            const error: UserPreferenceError = {
-                id:
-                    this.user?.userType === UserType.GROUP
-                        ? UserPreferenceErrorId.GROUP
-                        : UserPreferenceErrorId.UNAUTHENTICATED
-            };
-            return err(error);
+            const hasWidgetConfigurationChanges =
+                prefValues[PreferenceKey.WIDGET_CONFIGURATIONS] && Object.keys(prefValues).length === 1;
+            if (hasWidgetConfigurationChanges) {
+                return this._updatePrefs(prefValues);
+            } else {
+                this.informationBar.show('settings-auth');
+                const error: UserPreferenceError = {
+                    id:
+                        this.user?.userType === UserType.GROUP
+                            ? UserPreferenceErrorId.GROUP
+                            : UserPreferenceErrorId.UNAUTHENTICATED
+                };
+                return err(error);
+            }
         } else {
-            const keys = Object.keys(prefValues) as PreferenceKey[];
-            const cookie = this.cookies.read(USER_PREFERENCES_COOKIE_NAME);
-            const cookieValues = cookie ? JSON.parse(cookie) : {};
-            let updatedCookie = false;
-
-            keys.forEach((key) => {
-                if (COOKIE_PREFERENCES.includes(key)) {
-                    updatedCookie = true;
-                    cookieValues[key] = prefValues[key];
-                } else if (LOCALSTORAGE_PREFERENCES.includes(key)) {
-                    const value = JSON.stringify(prefValues[key]);
-                    if (!this.fastboot.isFastBoot) {
-                        localStorage.setItem(`${USER_PREFERENCES_LS_PREFIX}_${key}`, value);
-                    }
-                }
-            });
+            return this._updatePrefs(prefValues);
+        }
+    }
 
-            if (updatedCookie) {
-                const newCookie = JSON.stringify(cookieValues);
-                this.cookies.write(USER_PREFERENCES_COOKIE_NAME, newCookie, {
-                    sameSite: ENV.cookieSameSite,
-                    maxAge: MAX_AGE,
-                    secure: ENV.cookieSecure
-                });
+    /**
+     * Private update prefs method - not to be used by other classes
+     *
+     * @param {PreferenceChangeset} prefValues
+     * @return {*}  {(Promise<Result<UserPreferences | undefined, UserPreferenceError>>)}
+     * @memberof CurrentUserService
```

Comment:
```suggestion
     * @memberof CurrentUserService
     * @private
```
JSDoc has a `@private` tag it looks like https://jsdoc.app/tags-private.html

Reply:
Yea Im not concerned about that for now. It looks like truly private methods are coming in TS 4.3 in which case that will probably be auto generated when I use it.